test/parallel/test-tls-generic-stream.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const fixtures = require('../common/fixtures');
+const makeDuplexPair = require('../common/duplexpair');
+const assert = require('assert');
+const { TLSSocket, connect } = require('tls');
+
+const key = fixtures.readKey('agent1-key.pem');
+const cert = fixtures.readKey('agent1-cert.pem');
+const ca = fixtures.readKey('ca1-cert.pem');
+
+const { clientSide, serverSide } = makeDuplexPair();
+
+const clientTLS = connect({
+  socket: clientSide,
+  ca,
+  host: 'agent1'  // Hostname from certificate
+});
+const serverTLS = new TLSSocket(serverSide, {
```

Comment:
According to the docs a `secureConnect` event should be emitted once the connection is secure. However it does not work properly, as reported [here](https://github.com/nodejs/node/issues/10555) and [here](https://github.com/nodejs/node/issues/13368), and demoed [here](https://github.com/alexfernandez/tls-server-demo).  Maybe it is a good time to document it properly and test for it?

Reply:
Posting on a random pull request that was merged months ago isn't very helpful... can you move the discussion to #10555?

Reply:
Sure! Sorry about that, I commented here since @addaleax posted this PR to a discussion about https://github.com/nodejs/node/issues/13368.